src/Libraries/Module/ModuleManager.php

```diff
@@ -141,33 +145,48 @@ public function addModuleConfig()
      */
     public function writeContents()
     {
-        $this->copyDirectoryWithTemplates("$this->templatePath" . DS . "src", $this->modulePath);
+        $copiedFiles = [];
+        $this->copyDirectoryWithTemplates($this->templatePath . DS . "src", $this->modulePath, $copiedFiles);
 
         if ($this->withAssets) {
-            $this->copyAssets("$this->templatePath" . DS . "assets", $this->assetsPath);
+            $this->copyAssets($this->templatePath . DS . "assets", $this->assetsPath, $copiedFiles);
+        }
+
+        if (!$this->validateModuleFiles($copiedFiles)) {
+            throw new Exception("Module creation incomplete: missing files.");
         }
     }
 
     /**
+     * @param string $src
+     * @param string $dst
+     * @param array $copiedFiles
      * @throws Exception
      */
-    private function copyDirectoryWithTemplates(string $src, string $dst)
+    private function copyDirectoryWithTemplates(string $src, string $dst, array &$copiedFiles)
```

Comment:
If possible, let's try to avoid of having `$copiedFiles` to be passed here by reference and add results to it, instead let's make it so function returns results and assigns it to `$copiedFiles` to be more explicit